numba_dpex/core/runtime/_dpexrt_python.c

```diff
@@ -1328,7 +1334,9 @@ static int DPEXRT_sycl_event_from_python(PyObject *obj,
         goto error;
     }
 
-    event_struct->parent = obj;
+    Py_INCREF(event_obj);
```

Comment:
Please add a comment here. The `Py_INCREF` is to ensure that Numba lifetime analysis and cleanups do not impact the original Python object that was passed into Numba.